src/flush.rs

```diff
@@ -94,18 +96,22 @@ impl DbInner {
         self: &Arc<Self>,
         mut rx: UnboundedReceiver<FlushMsg<WalFlushThreadMsg>>,
         tokio_handle: &Handle,
-    ) -> Option<tokio::task::JoinHandle<()>> {
+    ) -> Option<tokio::task::JoinHandle<Result<(), SlateDBError>>> {
         let this = Arc::clone(self);
-        Some(tokio_handle.spawn(async move {
+        let fut = async move {
```

Comment:
yeah that makes sense. I wanted to leave it inline so the diff is obvious. It'll definitely be cleaner, and less error prone (it's easy to accidentally do a `return` instead of a `break`).